File: 프로그래머스/unrated/250137. ［PCCP 기출문제］ 1번 ／ 붕대 감기/［PCCP 기출문제］ 1번 ／ 붕대 감기.java
class Solution {
    public int solution(int[] bandage, int health, int[][] attacks) {
        int answer = health;
        int index = 0;
        int band_cnt = 0;
        for(int i=attacks[0][0];i<=attacks[attacks.length-1][0];i++){
            if(i==attacks[index][0]){
                band_cnt=0;
                answer-=attacks[index][1];
                index++;
                if(answer<1) return -1;
            } else{
                if(answer<health){
                    answer+=bandage[1];
                    band_cnt +=1;
                    if(band_cnt==bandage[0]){
                        band_cnt =0;
                        answer+=bandage[2];
                    }
                    if(answer>health) answer=health;
                }
                
            }
        }
        return answer;
    }
    
    
}

File: 프로그래머스/unrated/250137. ［PCCP 기출문제］ 1번 ／ 붕대 감기/［PCCP 기출문제］ 1번 ／ 붕대 감기.py
def solution(bandage, health, attacks):
    answer = health
    index = 0
    band_cnt = 0
    for i in range(attacks[0][0],attacks[len(attacks)-1][0]+1):
        if i==attacks[index][0]:
            band_cnt = 0
            answer-=attacks[index][1]
            if answer <1 :
                return -1;
            else :
                index+=1
        else :
            if answer<health:
                answer+=bandage[1]
                band_cnt+=1
                if band_cnt ==bandage[0]:
                    band_cnt = 0
                    answer+=bandage[2];
                if answer>health:
                    answer=health            
    return answer